src/extensions/layout-selector/index.js

```diff
@@ -117,6 +140,80 @@ class LayoutSelector extends Component {
 		} );
 	}
 
+	detectImageBlocks( clientIds ) {
+		const {
+			getBlockName,
+			getBlockAttributes,
+		} = this.props;
+
+		const imageBlockTypes = [ 'core/image', 'core/cover' ];
+		const galleryBlockTypes = [ 'core/gallery' ];
+
+		return clientIds.map( ( clientId ) => {
+			const blockName = getBlockName( clientId );
+			const blockAttributes = getBlockAttributes( clientId );
+
+			if ( imageBlockTypes.includes( blockName ) ) {
+				return { [ clientId ]: pick( blockAttributes, [ 'id', 'url' ] ) };
+			}
+
+			if ( galleryBlockTypes.includes( blockName ) ) {
+				return { [ clientId ]: pick( blockAttributes, [ 'ids', 'images' ] ) };
+			}
+		} );
+	}
+
+	uploadExternalImages( clientId, blockAttributes ) {
+		const {
+			getBlockAttributes,
+			mediaUpload,
+			updateBlockAttributes,
+		} = this.props;
+
+		const urls = [];
+
+		if ( blockAttributes.hasOwnProperty( 'images' ) ) {
+			blockAttributes.images.forEach( ( image ) => urls.push( image.url ) );
+		} else {
+			urls.push( blockAttributes.url );
+		}
+
+		urls.forEach( ( imageUrl, index ) => {
+			window.fetch( imageUrl )
```

Comment:
I would be curious if we could use `globalThis` instead of window. It would make the code a little more platform agnostic. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/globalThis

Reply:
I believe we can use globalThis. Although IE 11 doesn't have support it looks like babel handels it.

Reply:
Do we officially support IE though?

Reply:
Yes, we support IE >= 11 just like the [Gutenberg project](https://github.com/WordPress/gutenberg/blob/master/packages/browserslist-config/index.js). @richtabor would know if this has changed.

Reply:
Okay, if babels handles it let's use it then